hazelcast/src/main/java/com/hazelcast/internal/networking/Channel.java

```diff
@@ -105,12 +120,58 @@
      *
      * <p>Not thread safe. Should be called in channel writer thread.</p>
      *
+     * Method will be removed once the TLS integration doesn't rely on subclassing this channel.
+     *
      * @throws IOException
      */
     void closeOutbound() throws IOException;
 
     /**
-     * @see java.nio.channels.SocketChannel#close()
+     * Closes the Channel.
+     *
+     * When the channel already is closed, the call is ignored.
      */
     void close() throws IOException;
+
+    /**
+     * Checks if this Channel is closed. This method is very cheap to make.
+     *
+     * @return true if closed, false otherwise.
+     */
+    boolean isClosed();
+
+    /**
+     * Adds a ChannelCloseListener.
+     *
+     * @param listener the listener to register.
+     * @throws NullPointerException if listener is null.
+     */
+    void addCloseListener(ChannelCloseListener listener);
+
+    /**
+     * Checks if this is the 'client' side of a Channel or 'server' side.
+     *
+     * Each channel has a client/server side. The client side is the one that initiated the channel, and the serverside
+     * is the one that 'accepts' the channel.
+     *
+     * @return
+     */
+    boolean isClient();
```

Comment:
`isClient()` name may be confusing,  like Hazelcast client and server. What about something similar to `isClientMode()`? or any other more explicit name...